OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs

```diff
@@ -101,5 +101,26 @@ protected virtual bool ShouldFlee(Squad squad, Func<IEnumerable<Actor>, bool> fl
 
 			return flee(enemyAroundUnit);
 		}
+
+		protected static bool IsRearm(Actor a)
+		{
+			if (a.IsIdle)
+				return false;
+
+			var activity = a.CurrentActivity;
+			var type = activity.GetType();
```

Comment:
Can we be sure that the `CurrentActivity` is never `null` here?

Reply:
I will looks into it tomorrow.  This function is moved from `AirState` so I assume it is right.

Reply:
Hm, we have to be careful here. It might be that it only works for aircraft since planes always run an activity (flying idle in circles instead of doing nothing).

Reply:
`protected static bool BusyAttack(Actor a)` has the same problem if we see here. It does not check null when applying the similar activity check.

Reply:
Oh, here:
`public bool IsIdle { get { return CurrentActivity == null; } }`

Reply:
So this is null check:
`if (a.IsIdle)`